packages/mobile/src/exchange/saga.ts

```diff
@@ -167,9 +168,10 @@ export function* exchangeGoldAndStableTokens(action: ExchangeTokensAction) {
   Logger.debug(TAG, `Exchanging ${makerAmount.toString()} of token ${makerToken}`)
   let context: TransactionContext | null = null
   try {
-    navigate(Screens.ExchangeHomeScreen) // Must navigate to final screen before getting unlocked account which prompts pin
     ValoraAnalytics.track(CeloExchangeEvents.celo_exchange_start)
     const account: string = yield call(getConnectedUnlockedAccount)
+    navigate(Screens.ExchangeHomeScreen)
```

Comment:
Did me move this to after account being unlocked to allow enter PIN again if user has cancelled previous one?